_includes/tutorials/schedule-ktable-ttl/kstreams/markup/test/make-topology-test.adoc

```diff
@@ -0,0 +1,99 @@
+////
+  This content file is used to describe how to add test code you developed in this tutorial.  You'll need to update the
+  text to suit your test code.
+
+
+////
+
+Testing a Kafka streams application requires a bit of test harness code, but happily the `org.apache.kafka.streams.TopologyTestDriver` class makes this much more pleasant that it would otherwise be.
+
+////
+             You'll want to update the name of the test method from "exampleTest" to something more meaningful for your tutorial
+////
+
+There is only one method in `KafkaStreamsKTableTTLExampleTest` annotated with `@Test`, and that is `shouldTriggerStreamTableJoinFromTable()`. This method actually runs our Streams topology using the `TopologyTestDriver` and some mocked data that is set up inside the test method.
+
+
+This test is fairly vanilla, but there is one section we should look into a little more
+
+[source, java]
+----
+    final KafkaStreamsKTableTTLExample instance = new KafkaStreamsKTableTTLExample();
+    final Properties envProps = instance.loadEnvProperties(TEST_CONFIG_FILE);
+
+    final Properties streamProps = instance.getStreamProps(envProps);
+    final String inputTopicName = envProps.getProperty("input.topic.name");
+    final String outputTopicName = envProps.getProperty("output.topic.name");
+    final String tableTopicName = envProps.getProperty("table.topic.name");
+
+    final Topology topology = instance.buildTopology(envProps);
+    System.out.println(topology);
+    try (final TopologyTestDriver testDriver = new TopologyTestDriver(topology, streamProps)) {
+
+
+      final Serializer<String> keySerializer = Serdes.String().serializer();
+      final Serializer<String> valueSerializer = Serdes.String().serializer();
+      
+      final Deserializer<String> keyDeserializer = Serdes.String().deserializer();
+      final Deserializer<String> valueDeserializer = Serdes.String().deserializer();
+
+      
+      final TestInputTopic<String, String>  inputTopic = testDriver.createInputTopic(inputTopicName,
+                                                                                        keySerializer,
+                                                                                        valueSerializer);
+      final TestInputTopic<String, String>  tableTopic = testDriver.createInputTopic(tableTopicName,
+          keySerializer,
+          valueSerializer);
+
+      final TestOutputTopic<String, String> outputTopic = testDriver.createOutputTopic(outputTopicName, keyDeserializer, valueDeserializer);
+      final TestOutputTopic<String, String> outputTableTopic = testDriver.createOutputTopic(tableTopicName, keyDeserializer, valueDeserializer);
+
```

Comment:
Maybe leave this section out as the code below shows how the test is runs and the annotations describe what's going on.

Reply:
done